### ISSUE TITLE
Proposal: Don't build and deploy contrib stuff by default

### ISSUE THREAD

**Issue by krancour** [kubernetes-incubator/service-catalog#371]
Everything that's been relegated to `contrib/` has been so because it's been deemed "non-essential." Consequently, not everyone cares about the stuff that is in there. I'm not sure why, then, we're building, testing, and deploying those components by default.

I'd like to propose that the build be amended such that these components can be _optionally_ built (opt-in; not opt-out) _and_ that these components be separated into their own, optional, chart.

![Discuss](https://3.bp.blogspot.com/-fwdIjWK_gv4/VcmCiS8mRdI/AAAAAAAAZ0k/rfks8pRwFRo/s1600/Coffee%2BTalk%2B-%2BDiscuss.jpg)

P.S.: Obviously, CI should opt-in to building and testing _everything_.

**Comment by arschles** [comment]
I'm good with optional and opt-in

**Comment by pmorie** [comment]
SGTM, do it

On Mon, Feb 13, 2017 at 4:30 PM, Aaron Schlesinger <notifications@github.com

**Comment by duglin** [comment]
what do you mean by optional build?  While I'm ok with them being under their own Makefile target, do I think that our PR/CI system should build everything.

**Comment by krancour** [comment]
@duglin:

> Obviously, CI should opt-in to building and testing everything.

What I'm trying to accomplish here is reduced dependency on stuff in `contrib` for a few reasons:

1. Build, test, etc. currently builds and tests all the stuff in `contrib`-- which is stuff that many of us don't care too much about. This ends up being a big drag, locally, when hacking on the apiserver or controller.
1. Service catalog apiserver / controller Helm chart currently _also_ installs sample brokers... this is flat out wrong. "In real life," operators would install select brokers via each broker's _own_ Helm chart. Brokers wouldn't come packaged with the service catalog apiserver / controller. If that's not how it works in real life, that's not how our chart should be right now either.

Bottom line... this is a play at building / testing / _deploying_ less stuff... unless for some reason you care about those things (CI will care). Maybe, longer term, this is a step toward eliminating some of what's in contrib (e.g. sample brokers) and moving them off into their own repos where, if they are _useful_ brokers, they can have their own life.

**Comment by arschles** [comment]
This should be done by stable. Assigning to `1.0.0`

**Comment by duglin** [comment]
I think its important to distinguish between a dev build and a CI build. I can agree that a normal dev build might skip optional stuff because a dev's time is $.  However, if its in the repo then a full build (ie. CI/travis/...) should build/test it - otherwise things will get stale and we'll have stuff in the repo that will be broken or incorrect and that makes us look sloppy and will annoying people looking at our stuff.  If we don't care enough to want to build/test it to ensure it continually works then remove it from the repo.

**Comment by MHBauer** [comment]
Default make does build, test, verify.
https://github.com/kubernetes-incubator/service-catalog/blob/v0.1.21/Makefile#L15

At the time we got 5 binaries from a default `make`
https://github.com/kubernetes-incubator/service-catalog/blob/acce1d478099c5d2fcd7ee6beb03dbf9fbef379d/Makefile#L74-L76

Now we have 3 outputs:
https://github.com/kubernetes-incubator/service-catalog/blob/v0.1.21/Makefile#L122-L125

We added more stuff to contrib. But it's not built or checked by default. I think it's successfully cleaner now. @krancour
